### ISSUE TITLE
Regression: "Already imported a file that will be measured", loss of coverage

### ISSUE THREAD

**Issue by mgedmin** [coveragepy/coveragepy#909]
**Describe the bug**

I'm measuring [gtimelog's test coverage](https://coveralls.io/github/gtimelog/gtimelog) with `coverage run -m gtimelog.tests`.  Since coverage 5.0a4 this is missing some coverage (shows 79% instead of 100%) and also produces warnings:
```
Coverage.py warning: Already imported a file that will be measured: /home/mg/src/gtimelog/src/gtimelog/__init__.py (already-imported)
Coverage.py warning: Already imported a file that will be measured: /home/mg/src/gtimelog/src/gtimelog/timelog.py (already-imported)
Coverage.py warning: Already imported a file that will be measured: /home/mg/src/gtimelog/src/gtimelog/settings.py (already-imported)
```

**To Reproduce**
How can we reproduce the problem?  Please *be specific*.
1. What version of Python are you using? 2.7.17 and 3.7.5
1. What version of coverage.py are you using? 5.0.1 (but every version starting with 5.0a4 is affected; 5.0a3 is fine)
1. What versions of what packages do you have installed? 

    <details>

    ```
    coverage==5.0.1
    coverage-python-version==0.2.0
    freezegun==0.3.12
    -e git+https://github.com/gtimelog/gtimelog@5f7858cc58925e42b2d7cbc1d872b11745c38872#egg=gtimelog
    mock==3.0.5
    python-dateutil==2.8.1
    six==1.13.0
    </details>

1. What code are you running?  Give us a specific commit of a specific repo that we can check out.
https://github.com/gtimelog/gtimelog/tree/5f7858cc58925e42b2d7cbc1d872b11745c38872
1. What commands did you run? `make coverage` for a quick version; `.tox/coverage/bin/pip install coverage=$version && .tox/coverage/bin/coverage -m gtimelog.test && .tox/coverage/bin/coverage report` to test a specific version of coverage.py

**Expected behavior**
100% coverage reported, no warnings.

**Comment by mgedmin** [comment]
git bisect blames commit b7e0eec3d794ee350f8a4392a0c9cc90fff54fa0

**Comment by mgedmin** [comment]
So it looks like `coverage run -m gtimelog.tests` imports `gtimelog/__init__.py` and `gtimelog/tests/__init__.py` before starting coverage measurement.  As a workaround I can apply this patch to my (hacky) test code:

```
diff --git a/src/gtimelog/tests/__init__.py b/src/gtimelog/tests/__init__.py
index 02f7cc5..31f8aca 100644
--- a/src/gtimelog/tests/__init__.py
+++ b/src/gtimelog/tests/__init__.py
@@ -1,10 +1,9 @@
 """Tests for gtimelog"""
 import unittest
 
-from gtimelog.tests import test_timelog, test_settings, test_main
-
 
 def test_suite():
+    from gtimelog.tests import test_timelog, test_settings, test_main
     return unittest.TestSuite([
         test_timelog.test_suite(),
         test_settings.test_suite(),
```

and then I have only one wrongly reported missing code line: the `__version__` assignment in `gtimelog/__init__.py`.

I would prefer not having to do that: after all, things worked fine with coverage 4.x (or 5.0a3).

(And anyway the real workaround would be to stop being silly with custom test running modules and start using pytest, which is on my todo list anyway.)

**Comment by nedbat** [comment]
This is now available in [coverage 5.0.2](https://pypi.org/project/coverage/5.0.2/)
